File: 0036-valid-sudoku/0036-valid-sudoku.js
/**
 * @param {character[][]} board
 * @return {boolean}
 */

var check_row=function(i,j,board){

for(let k=0;k<9;k++){
    if(k==j){
        continue;
    }
    if(board[i][k]==board[i][j]){
        return false;
    }
  
}
  return true;
}
var check_col=function(i,j,board){
    
for(let k=0;k<9;k++){
    if(k==i){
        continue;
    }
    if(board[k][j]==board[i][j]){
        return false;
    }
   
}
  return true;   
}
var check_box=function(i,j,board){
let a= Math.floor(i/3)*3;
let b= Math.floor(j/3)*3;
for(let x=a;x<a+3;x++){
    for(let y=b;y<b+3;y++){
        if(x==i && y==j){
            continue;
        }
        if(board[x][y]==board[i][j]){
            return false;
        }
    }
}
return true;

}
var is_safe=function( i, j,board){
    return check_row(i,j,board) && check_col(i,j,board) && check_box(i,j,board);
}
var isValidSudoku = function(board) {
    
for(let i=0;i<9;i++){
    for(let j=0;j<9;j++){
        if(board[i][j]=='.'){

        }else if(is_safe(i,j,board)){

        }else{
            return false;
        }
    }
}

return true;

};

File: 0036-valid-sudoku/0036-valid-sudoku.cpp
class Solution {
public:

    bool check_box(int i, int j,vector<vector<char>>& board){
        
        unordered_map<char,int> temp;
        for(int n=i;n<i+3;n++){
            for(int m=j;m<j+3;m++){
                if(board[n][m] == '.') continue;
                temp[board[n][m]]++;

                if(temp[board[n][m]]>=2){
                    return false;
                }
            }
        }
        return true;
    }

    bool check_row(int i,vector<vector<char>>& board){

             unordered_map<char,int> temp;
            for(int n=0;n<=8;n++){
                if(board[i][n] == '.') continue;
                temp[board[i][n]]++;
                 if(temp[board[i][n]]>=2){
                    return false;
                }
            }
    return true;
    }

    bool check_col( int j,vector<vector<char>>& board){

         unordered_map<char,int> temp;
            for(int n=0;n<=8;n++){
                if(board[n][j] == '.') continue;
                temp[board[n][j]]++;
                 if(temp[board[n][j]]>=2){
                    return false;
                }
            }
    return true;

    }

    bool isValidSudoku(vector<vector<char>>& board) {

           

        for(int i=0;i<9;i++){
            if(!check_col(i,board))return false;
            if(!check_row(i,board))return false;

            int k=(i/3)*3;
            int l=(i%3)*3;
            if(!check_box(k,l,board))return false;
        }
        return true;
    }
};